Patch from Kyle Moffett <kyle@moffetthome.net>

Update GnuTLS support to allow specifying multiple ciphers as a
colon-separated list in TLSCipherSuite.

Addresses ITS#5341.

Index: trunk/libraries/libldap/tls.c
===================================================================
--- trunk.orig/libraries/libldap/tls.c
+++ trunk/libraries/libldap/tls.c
@@ -300,6 +300,7 @@
 		for (i=0; i<n_ciphers; i++) {
 			if ( !strncasecmp( ciphers[i].name, ptr, len )) {
 				num++;
+				ptr = end + 1;
 				break;
 			}
 		}
@@ -330,6 +331,7 @@
 			 * only appear once in each list.
 			 */
 			if ( !strncasecmp( ciphers[i].name, ptr, len )) {
+				ptr = end + 1;
 				for (j=0; j<nkx; j++)
 					if ( kx[j] == ciphers[i].kx )
 						break;
